src/library-authoring/collections/CollectionDetails.tsx

```diff
@@ -0,0 +1,154 @@
+import { FormattedMessage, useIntl } from '@edx/frontend-platform/i18n';
+import { Icon, Stack } from '@openedx/paragon';
+import { useContext, useState } from 'react';
+import classNames from 'classnames';
+
+import { getItemIcon } from '../../generic/block-type-utils';
+import { ToastContext } from '../../generic/toast-context';
+import { BlockTypeLabel, type CollectionHit, useSearchContext } from '../../search-manager';
+import type { ContentLibrary } from '../data/api';
+import { useUpdateCollection } from '../data/apiHooks';
+import HistoryWidget from '../generic/history-widget';
+import messages from './messages';
+
+interface BlockCountProps {
+  count: number,
+  blockType?: string,
+  label: React.ReactNode,
+  className?: string,
+}
+
+const BlockCount = ({
+  count,
+  blockType,
+  label,
+  className,
+}: BlockCountProps) => {
+  const icon = blockType && getItemIcon(blockType);
+  return (
+    <Stack className={classNames('text-center', className)}>
+      <span className="text-muted">{label}</span>
+      <Stack direction="horizontal" gap={1} className="justify-content-center">
+        {icon && <Icon src={icon} size="lg" />}
+        <span>{count}</span>
+      </Stack>
+    </Stack>
+  );
+};
+
+const CollectionStatsWidget = () => {
+  const {
+    blockTypes,
+  } = useSearchContext();
+
+  const blockTypesArray = Object.entries(blockTypes)
+    .map(([blockType, count]) => ({ blockType, count }))
+    .sort((a, b) => b.count - a.count);
+
+  const totalBlocksCount = blockTypesArray.reduce((acc, { count }) => acc + count, 0);
+  const otherBlocks = blockTypesArray.splice(3);
+  const otherBlocksCount = otherBlocks.reduce((acc, { count }) => acc + count, 0);
+
+  if (totalBlocksCount === 0) {
+    return (
+      <div className="text-center text-muted">
+        <FormattedMessage {...messages.detailsTabStatsNoComponents} />
+      </div>
+    );
+  }
+
+  return (
+    <Stack direction="horizontal" className="p-2 justify-content-between" gap={2}>
+      <BlockCount
+        label={<FormattedMessage {...messages.detailsTabStatsTotalComponents} />}
+        count={totalBlocksCount}
+        className="border-right"
+      />
+      {blockTypesArray.map(({ blockType, count }) => (
+        <BlockCount
+          key={blockType}
+          label={<BlockTypeLabel type={blockType} />}
```

Comment:
We also have one endpoint ([mock here](https://github.com/openedx/frontend-app-authoring/blob/3662fadad44a610c95f6169760d48df8305cd6a0/src/library-authoring/data/api.mocks.ts#L8-L43)) that returns this same data (but not regionalized).

Could we handle this in a future task?

Reply:
> Could we handle this in a future task?

Yep, this can totally be in a separate task -- might have to push back on UI/UX review of this task, because it's a "nice to have" on this label, and not that simple to implement.

But that endpoint you referenced returns the [allowed block types for a given content library](https://github.com/openedx/edx-platform/blob/dac0309b0ff590ebaba5cf52e8fe518bd0c38270/openedx/core/djangoapps/content_libraries/views.py#L483), which I suspect is an old use case for libraries v2, since we create libraries that support all block types (with [lib.type=='COMPLEX'](https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/content_libraries/models.py#L114) by default), and there's nowhere in the frontend to change that. I personally don't think we should be using a REST API to tell us what different block types are called -- the app translations are enough.

Reply:
Eventually we can update `BlockTypeLabel` so that it can load and cache the translated block names for random third-party XBlocks, but it definitely shouldn't bother using a REST API for the "standard" block types.

> But that endpoint you referenced returns the [allowed block types for a given content library](https://github.com/openedx/edx-platform/blob/dac0309b0ff590ebaba5cf52e8fe518bd0c38270/openedx/core/djangoapps/content_libraries/views.py#L483), which I suspect is an old use case for libraries v2, since we create libraries that support all block types (with [lib.type=='COMPLEX'](https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/content_libraries/models.py#L114) by default)

Exactly. That endpoint should be deprecated and is not the right one to use if it's not localized.